frontend/src/components/LoxotronCasinoGameComponent/LoxotronCasinoSlotMachine/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { useRef, useEffect } from 'react';
+import frameUrl from '../../../assets/svg/Frame.svg';
+import './slotMachine.sass'
+import svg1 from '../../../assets/svg/tiles/1.svg';
+import svg2 from '../../../assets/svg/tiles/2.svg';
+import svg3 from '../../../assets/svg/tiles/3.svg';
+import svg4 from '../../../assets/svg/tiles/4.svg';
+import svg5 from '../../../assets/svg/tiles/5.svg';
+import svg6 from '../../../assets/svg/tiles/6.svg';
+import svg7 from '../../../assets/svg/tiles/7.svg';
+import svg8 from '../../../assets/svg/tiles/8.svg';
+import svg9 from '../../../assets/svg/tiles/9.svg';
+import svg10 from '../../../assets/svg/tiles/10.svg';
+
+
+const SlotMachineComponent = () => {
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+
+    useEffect(() => {
+        if (!canvasRef.current) return;
+
+        const canvas = canvasRef.current as HTMLCanvasElement;
+        const context = canvas.getContext("2d");
+        const screenWidth = 1572;
+        const screenHeight = 622;
+
+        canvas.width = screenWidth;
+        canvas.height = screenHeight;
+
+        abstract class DrawableObject {
+            abstract draw(x: number, y: number): void;
+        }
+
+        class FillObject extends DrawableObject {
+            protected width: number = 1480;
+            protected height: number = 11;
+            protected x: number = 46;
+
+            draw(y: number) {
+                if(context) {
```

Comment:
Are you sure that it's not null? I prefer to have context as a static variable for DrawableObject and then all children will use it.